netket/hilbert/spin.py

```diff
@@ -123,8 +121,16 @@ def _check_total_sz(self, total_sz, size):
                 "Cannot fix the total magnetization: Nspins + " "totalSz must be even."
             )
 
+    def __pow__(self, n):
+        if self._total_sz is None:
+            total_sz = None
+        else:
+            total_sz * n
```

Comment:
There is an `=` missing. (If the tests pass, we should probably add a test for power with constraint, as this line is apparently not covered.)